OnePasswordSource.m

```diff
@@ -119,36 +84,42 @@ - (BOOL)indexIsValidFromDate:(NSDate *)indexDate forEntry:(NSDictionary *)theEnt
 
 
 - (BOOL)loadChildrenForObject:(QSObject *)object {
-	// For the children to 1Pwd, just load what's in objectsForEntry
-	if([[object primaryType] isEqualToString:QSFilePathType]) {
-		NSArray *items = [self objectsForEntry:[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"LoadingChildren"]];
-		if (!items) {
-			return NO;
-		}
-		[object setChildren:items];
-		return YES;
-	}
-	return NO;
+	// For the children to 1Pwd, just load items from the catalog
+	NSMutableArray *children = [[QSLib scoredArrayForType:QS1PasswordForm] mutableCopy];
+	[object setChildren:children];
+	return YES;
 }
 
-- (NSArray *)objectsForEntry:(NSDictionary *)theEntry{
+- (NSArray *)objectsForEntry:(QSCatalogEntry *)theEntry
+{
 	// Define the objects (Empty to start with) we're going to send back to QS
 	NSMutableArray *objects = [NSMutableArray array];
 	QSObject *newObject;
-	NSString *location = [[[NSUserDefaults standardUserDefaults] objectForKey:k1PPath] stringByStandardizingPath];
-	NSData *JSONData = [NSData dataWithContentsOfFile:location];
-	// TODO - check to make sure this is valid JSON data before running yajl_JSON OR switch to NSJSONSerialization
-	NSArray *OPItems = [JSONData yajl_JSON];
-	for (NSArray *metadata in OPItems) {
-		NSString *uuid = metadata[0];
-		NSString *title = metadata[1];
-		NSString *location = metadata[2];
-		newObject = [QSObject makeObjectWithIdentifier:[NSString stringWithFormat:@"%@-%@", location, uuid]];
+	NSString *location = [self keychainPath];
+	NSSet *searchLocations = [NSSet setWithObject:location];
+	NSMetadataQuery *passwordSearch = [[NSMetadataQuery alloc] init];
+	NSArray *OPItems = [passwordSearch resultsForSearchString:@"kMDItemContentType == 'com.agilebits.itemmetadata'" inFolders:searchLocations];
+	for (NSMetadataItem *item in OPItems) {
+		NSDictionary *metadata = [item valuesForAttributes:@[(NSString *)kMDItemPath]];
+		NSString *itemPath = metadata[(NSString *)kMDItemPath];
+		NSData *JSONData = [NSData dataWithContentsOfFile:itemPath];
+		NSDictionary *OPItem = [JSONData yajl_JSON];
+		if (![OPItem[@"categorySingularName"] isEqualToString:@"Login"]) {
```

Comment:
recommend switching to categoryUUID here and filtering based on the list at the bottom of https://support.1password.com/integration-mac/